kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -212,6 +215,20 @@ class BatchJobSubmission(
         metadata match {
           case Some(metadata) if metadata.peerInstanceClosed =>
             setState(OperationState.CANCELED)
+          case Some(metadata)
+              // in case it has been updated by peer kyuubi instance, see KYUUBI-6278
```

Comment:
nit: `KYUUBI-6278` => `KYUUBI #6278`

does this fix another issue? what happens without this part of the code before/after this PR?

Reply:
Let me move it to another separate PR.

Reply:
The issue is that,

1.  if the batch state already updated by peer with terminated state
2. without this change, it would try to get the current application state and might NOT_FOUND

Reply:
Before this PR:

1. it might get NOT_FOUND
2. then batch finished

Reply:
After this PR:
1. respect the terminated application state set by peer, and then FINISH/FAIL the batch.

Reply:
FYI:
https://github.com/apache/kyuubi/blob/70c03ef998bb695d62202f2378f3a09c5fcdfd83/kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala#L323-L325

For monitorBatchJob, it would get the currentApplicationInfo first, and does not respect the metadata store application state.